cognite/neat/rules/models/entities.py

```diff
@@ -211,6 +213,11 @@ def as_non_versioned_entity(self) -> str:
             return f"{self.suffix!s}"
         return f"{self.prefix}:{self.suffix!s}"
 
+    def as_dms_compliant_entity(self) -> "Self":
+        new_entity = self.model_copy(deep=True)
+        new_entity.suffix = replace_non_alphanumeric_with_underscore(new_entity.suffix)
+        return new_entity
```

Comment:
This is elegant way to solve it. Any reason not to move it down to the `DMSEntity` class instead? 

Not sure why a general `Entity`, for example, `ClassEntity` needs to know about `DMS`

Reply:
Moved it to ClassEntity